prinz_h2o/src/main/scala/pl/touk/nussknacker/prinz/h2o/repository/H2OModelRepository.scala

```diff
@@ -1,3 +1,31 @@
 package pl.touk.nussknacker.prinz.h2o.repository
 
-class H2OModelRepository
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.prinz.h2o.H2OConfig
+import pl.touk.nussknacker.prinz.h2o.model.H2OModel
+import pl.touk.nussknacker.prinz.model.{Model, ModelName}
+import pl.touk.nussknacker.prinz.model.repository.{ModelRepository}
+import pl.touk.nussknacker.prinz.util.repository.{ModelPayload, RepositoryClient}
+
+class H2OModelRepository(implicit config: H2OConfig)
+  extends ModelRepository with LazyLogging {
+
+  private val client = new RepositoryClient
```

Comment:
instead of creating in every subclass new RepositoryClient we can create a trait (for example ModelClientRepository) in prinz-api which will have protected final field with RepositoryClient so writing "with ModelClientRepository" will replace creating the RepositoryClient  in every subclass - fo rme it looks more scala's styled